### ISSUE TITLE
Next beta version / feature release

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#10922]
As discussed in our team meeting we favor a beta version from `master` as soon as possible.
The plan is to publish a beta as soon as (from our perspective/knowledge) there are no major functional regressions.

In order to do that, we need to forward `release` to the HEAD of `master` which means afterwards:
- Development (e.g. bottom bar) on `master` can continue as soon as branches have been merged
- Fixes for bugs reported for the beta version (and other bugfixes) should then go to `release` instead of `master`. (If we follow up with another beta without releasing it would theoretically make no difference if we do this on `master` but as we might be forced to release that version due to unrelated regressions (website change) it still makes sense to follow the normal process).

We should use this issue to align, whether the current `master` is considered ready for beta state.
We need to crawl through the issues to see if there are known major regressions next days.

**Comment by moving-bits** [comment]
Just a small observation:

If you have a look at our [changelog for master](https://github.com/cgeo/cgeo/blob/master/main/res/raw/changelog_master.md) there is the usual bunch of documented changes for a version, quite a few actually. 

But aditionally, we have the changes our new filter system and for the UI changes, which are not documented in detail in the change log. If you look at the two respective [projects](https://github.com/cgeo/cgeo/projects), you can see that they alone close another 120 (!) or so issues - very impressive! 

Thanks for all your contributing work in graphics, coding, testing & discussing!

**Comment by fm-sys** [comment]
I think we should fix #10940 first and do another bugfix release, before we start into the new large beta...

**Comment by Lineflyer** [comment]
Yes, that might be wise.
Currently not so many support mail regarding this problem as it is no top-prio feature, still we should maybe fix it not only in some months (as design finalization + beta + bugfix might take).

**Comment by eddiemuc** [comment]
#10997 should fix the watchlist bug

**Comment by Lineflyer** [comment]
Fix has been merged and tested OK.
I will prepare a bugfix release now.


- [ ] Check/Insert correct version name as headline in ```changelog-release.xml``` 
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid 
- [ ] Trigger notification to status.cgeo.org via CI server once the release is available on Google Play
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by Lineflyer** [comment]
Release 2021.06.23 has been uploaded to Google Play.

**Comment by moving-bits** [comment]
Now that #10940 is fixed and the bugfix release is rolled out let us think about the planned "big beta" version: What needs to be done code-wise to start a new beta? Are there any urgent things left, or can we fast-forward, create a beta from current `master` state, and continue development (especially the bottom navigation) on `master`?

**Comment by eddiemuc** [comment]
Since SAF seems to be working good in production I would propose switching to Android 11 / SDK 30 with next beta/release. So we still have some time to react until September if we discover problems.

What do you think?

**Comment by eddiemuc** [comment]
Oh and additionally I would introduce a small PR removing the "experimental" from new filtering, renaming the old filtering to "legacy" and switching their positions. I feel confident enough for this to go into beta.

Apart from that I would vote for a beta rather quickly to enable introduction of navigation bar. So the spirit/flow of current big redesign can continue.

**Comment by fm-sys** [comment]
I have an internal ToDo list on my mobile, which I fill with things I notice while playing with the nightly. Often I don't create issues on github for these findings, because it's way faster to create a small PR next time I'm at my development machine.

Some findings:
- [ ] filters activity actionbar currently uses the "send" icon, which is inconsistent, as we only use this icon when something gets uploaded. Instead we should use the ✔️ instead. Furthermore the hint text is hardcoded/unlocalized. And there is no warning that changes will be dismissed when closing the filters UI with back button (as for routesort, waypointEdit, ...) 
![Screenshot_20210627_113816](https://user-images.githubusercontent.com/64581222/123539879-30b35200-d73c-11eb-9292-f91ec6c36ee6.jpg)

- [ ] issue #10977
- [ ] subtitles should be single-line
![Screenshot_20210627_112747_cgeo geocaching_copy_431x357](https://user-images.githubusercontent.com/64581222/123539598-d1087700-d73a-11eb-80f8-2621b3dcdcd5.png)
- [ ]  UDC cache icon looks a bit ugly and has white border (unrelated to redesign or other recent changes)
![Screenshot_20210627_113713_cgeo geocaching](https://user-images.githubusercontent.com/64581222/123539862-0feafc80-d73c-11eb-8fbc-cad871c0928b.png)


I will be out geocaching the whole day, so I can work on these findings at the earliest tonight or tomorrow. If someone else would like to fix those, feel free to do so ;-)

**Comment by fm-sys** [comment]
PRs are ready for all my findings, so from my side we can start the new beta as soon as all PRs are merged...

OK, admitted, there are quite some things left but as some point we need to do the cut ;-)

For #11006, I unfortunately haven't found a solution yet...

**Comment by Lineflyer** [comment]
Like the plan.
Lets finalize/discuss/merge the PRs and then just make a cut, revisit known regressions (either solve or work on it while first beta is out) and do a  first beta release.
We can then step by step adress new issues or add current WIP and do consecutive beta releases.

**Comment by murggel** [comment]
> PRs are ready for all my findings, so from my side we can start the new beta as soon as all PRs are merged...
> 
> OK, admitted, there are quite some things left but as some point we need to do the cut ;-)
> 
> For #11006, I unfortunately haven't found a solution yet...

I probably found a solution for #11006, will check for different API tomorrow

**Comment by eddiemuc** [comment]
Please wait for #11018. I have a fix, just need to clean up all the experimental code

**Comment by Lineflyer** [comment]
Time for a cut now?

@fm-sys Maybe we should fix #11063 as it seems to be an unwanted regression?

I do see 9 issues in "Short term UI-todo", not sure if there is short term WIP or not.
If you think we are ready for a beta just tell me.
To me it looks quite OK now and we could use the feedback from beta users to adjust some things (e.g. if users like or dislike the bigger text, etc.).

Still on vacation right now, but pushing a beta version in the evening should not be a problem. ;)

**Comment by eddiemuc** [comment]
Wrt filter stuff, you have a go from my side :-)

**Comment by moving-bits** [comment]
Let's wait for @murggel's work #11065, looks promising.

**Comment by murggel** [comment]
> Let's wait for @murggel's work #11065, looks promising.

PR #11070 is on its way...

**Comment by murggel** [comment]
How about #11060?

**Comment by eddiemuc** [comment]
I feel we have a lot of UI issues still under way, but all of them are rather small and IMHO should not prevent releasing a first beta Version. If everything goes as usual, we will have updates to the beta version anyway where we can add these UI updates piece by piece.

But I am really eager to get feedback wrt switch to SDK30 and the filter stuff.

**Comment by Lineflyer** [comment]
I agree.
@eddiemuc  How about removing the legacy filter completely? I dont think its needed anymore. (Beta) users shall rather be forced to use the new one.

**Comment by Lineflyer** [comment]
Another thing:
Could someone summarize the status about:
- Removing the global filter (AFAIR we agreed to do this)
- Removing the global hiding of certain caches and only have the map specific hiding + the generic filter options

**Comment by moving-bits** [comment]
Regarding the UI topics listed in [Short-term UI todos](https://github.com/cgeo/cgeo/projects/20): Most of them are fixed already, and the remaining ones are no blockers, IMHO. Some we may not even be able to fix, and others can be fixed throughout a beta phase so no reason to prevent starting a beta (which would give us valuable feedback on new filters, new UI, and SDK 30).

**Comment by eddiemuc** [comment]
> I agree.
> @eddiemuc  How about removing the legacy filter completely? I dont think its needed anymore. (Beta) users shall rather be forced to use the new one.

Ok, but before the beta I suggest to only remove the menu entry (and not yet the big amount of code actually **performing** the filtering) so we can easily turn it on again if needed. Would that be ok?

**Comment by eddiemuc** [comment]
> Another thing:
> Could someone summarize the status about:
> - Removing the global filter (AFAIR we agreed to do this?
> - Removing the global hiding of certain caches and only have the map specific hiding + the generic filter options (AFAIR we decided to go this way)

The short answer: we agreed on this, but there's a vast amount of work to do before this is possible.

The global filters usage (both cache type and map hidings) is spread through nearly all parts of the code, especially through nearly all connectors, and it can only be removed after I enabled the new filter for all those connectors. Unfortunately I can't currently invest the coding time to change all of this in the next few days.

The map viewer as well as the offline lists already work consistently. As for the online search I just changed the "keyword" search until now. Other searches (address, owner etc) will not consider the new filter yet. If I remove the old global filter stuff they won't consider anything.

**Comment by fm-sys** [comment]
Another thing: IIRC @Lineflyer had suggested to replace the global cache type filter with a info popup  how to use the  new filter framework.

**Comment by eddiemuc** [comment]
> Another thing: IIRC @Lineflyer had suggested to replace the global cache type filter with a info popup  how to use the  new filter framework. 

Feel free to open an issue for that :-)

**Comment by eddiemuc** [comment]
> I agree.
> @eddiemuc How about removing the legacy filter completely? I dont think its needed anymore. (Beta) users shall rather be forced to use the new one.

PR #11087 created

**Comment by Lineflyer** [comment]
Wow. Quite some activity of you guys here today. Looks a bit as if the most important things are done now (including removal of the legacy filter).

As I am still on holiday I don't have my PC with me. However I have git running  on my tablet. :)
So we could merge branches and push a beta version tonight.

Anything left to urgently fix or add before doing this?
Otherwise I will start the process at 21:00 CET tonight.

**Comment by fm-sys** [comment]
Just checked the statistics again. Nearly 500 (!) issues/PRs have been created since the first PR #10643. That's simply amazing. Keep fingers crossed that we haven't broken too much ;-)

> Looks a bit as if the most important things are done now (including removal of the legacy filter).

IMHO, the global cache type filter can be considered as legacy filter as well. I have just opened the PR #11099. I would vote for merging before the beta, so that users are really forced to try out the new filter system. @Lineflyer please tell me what you think about it.

However, feels like it's the right time for the first beta today...

**Comment by Lineflyer** [comment]
@fm-sys Yes, lets do that.

I will merge that PR and spend some time to play around with the resulting beta.
I will keep all updated in this issue.

**Comment by MagpieFourtyTwo** [comment]
[...remove legacy filter]
> Ok, but before the beta I suggest to only remove the menu entry (and not yet the big amount of code actually performing the filtering) so we can easily turn it on again if needed. Would that be ok?

What about legacy filter leftovers, i. e. filters which were still active while updating ...?
If even the filter indicator string below the headline directly leads to the new filter system,
there may be no way to at least clear legacy filters - would suggest to clear them or first start ...?

**Comment by eddiemuc** [comment]
> What about legacy filter leftovers, i. e. filters which were still active while updating ...?

Afair the legacy filter setting is not stored beyond the lifetime of the cache list activity

**Comment by eddiemuc** [comment]
> @fm-sys Yes, lets do that.
> 
> I will merge that PR and spend some time to play around with the resulting beta.
> I will keep all updated in this issue.

Several "search by..." Functions will not be filterable by cache type if global cache type filter is disabled (search by address, by owner, by finder), but apart from that there should be no regression. Wonder if users even notice.

**Comment by fm-sys** [comment]
> there may be no way to at least clear legacy filters - may be necessary to "automatically" clear them or first start ...?

The advanced legacy filter doesn't get persisted and for the global cache type filter we don't  use the old preference value but use All instead. Therefore, everything should be fine AFAIK.

> Several "search by..." Functions will not be filterable by cache type if global cache type filter is disabled (search by address, by owner, by finder), but apart from that there should be no regression. Wonder if users even notice.

This can still be done with the new filter framework. Only difference is that the filter is applied client side. Probably something users really don't notice...

**Comment by Lineflyer** [comment]
Branch merging has been done, so ˋreleaseˋ is not equal to ˋmasterˋ.
I did however not alter any of the changelogs (as we did before), because I have to think about the translation consequences if we move content from one markdown file to the other. That could mass trigger retranslations, which should be avoided.

Need to think about that.
For now we can leave it like that, but please do not modify any changelog file on either branch.
Besides from that you might start working on ˋmasterˋ after the beta is out (will notify).

**Comment by eddiemuc** [comment]
> This can still be done with the new filter framework. Only difference is that the filter is applied client side. 

Problem is that the old (still active) search just fetches 20 caches from GC.com. if this is filtered client-sided, the resulting cache count might be pretty low.
I tried to fast-implement the switch of the affected functions to.new filter framework today (see my open PR) but did not find the time to test and stabilize it unfortunately.

**Comment by Lineflyer** [comment]
I created #11101 regarding the changelog.
We could either just publish the beta anyway, the only drawback would be, that the changelog shows "Nightly build" as caption. But I will wait a day to think about it and maybe get feedback on that issue from @moving-bits

**Comment by Lineflyer** [comment]
As just discussed in an open PR:

We keep the working mode as usual:
Improvements should now go to `release` while new features (e.g. bottom navigation) can already be developed on `master`

But:
Please kindly wait until #11101 has been depuzzled. While it might be independent from other changes its still possible, that the solution might still involve another merging to create a beta version with correct changelog.

**Comment by eddiemuc** [comment]
Since beta is not out yet, I would ask for ##11094 to be included as well. Since we decided to remove GlobalFilter right now, this will change at least most of the online searches to new filter framework.

Note also new issue #11103 wrt search using "Found By"!

**Comment by fm-sys** [comment]
I would be fine with including https://github.com/cgeo/cgeo/pull/11094 as well.

But in that case I would strongly suggest to merge #11100 as well, as the current rating stars look a bit broken... The PR is totally uncritical and provides a way more clean optic. 

Old|new
-|-
![Screenshot_20210705_142419_cgeo geocaching_copy_278x85](https://user-images.githubusercontent.com/64581222/124471966-d5b5d680-dd9d-11eb-8acb-86ae8f61f217.png)|![IMG_20210705_143008](https://user-images.githubusercontent.com/64581222/124471789-98e9df80-dd9d-11eb-8f0e-d4e0ab36a6b3.png)

**Comment by Lineflyer** [comment]
We can include #11094 but then its pretty untested before. But fine for me, its a beta and we want testing feedback anyway.

Feel free to merge the two mentioned PRs to master. I will update branches again tonight before puzzling the changelogs.

**Comment by moving-bits** [comment]
merged both PR to `release`, so no new branching from `master` necessary

**Comment by eddiemuc** [comment]
Wow you are fast :-)

**Comment by Lineflyer** [comment]
I corrected the changelogs manually now and hopefully did it right to reach the desired state for now.
For most languages the changelog of the beta version will now be english, but anyway we can get it out and solve the changelog issue lateron (see dedicated issue).

Will proceed now to build RC version (and a new nightly) to review changelogs and smoke test.

**Comment by moving-bits** [comment]
Thanks @Lineflyer for your efforts!

Eagerly awaiting first user feedbacks from our new beta...

**Comment by Lineflyer** [comment]
Took a look at the RC/beta and it looks as desired. As written above the only downside is missing translations for most languages, but it shows up properly with version code and english content at least.

Base functions also look normal. Will push it to beta users now and report back.

@moving-bits Will try to monitor channels next days but my availability is limited on vacation.If you read anything critical, feel free to open an issue. Otherwise I will read through support mails next week.

**Comment by Lineflyer** [comment]
Beta release 2021.07.05-RC has been uploaded and will be available to beta testers via Google Play soon.
No problems/warning while uploading in Google checks.

**Comment by Lineflyer** [comment]
I have seen the NPE issue in low power mode (which is already fixed).
I crawled through support mails and this problem is the only one reported frequently + what I transfered into #11118 

Additionally there are some users claiming some searches are broken or return wrong results or results without proper filtering. For all of those I first asked for a step-by-step description as the description was not preise enough to reproduce, which I did not receive yet.

Still I would like to push another update for the NPE fix rather soon.
Shall I go for another round tonight to get rid of the NPE problem or anything else already "in the pipe"?

**Comment by Lineflyer** [comment]
Will go for another beta 2021.07.06-RC. Working on it now.

**Comment by Lineflyer** [comment]
New beta 2021.07.06-RC has been uploaded and pushed out as beta version.

**Comment by Lineflyer** [comment]
Many issues fixed sine last beta. While the filtering clarification is still ongoing I would still like to push a new beta version right now, as we improved the text size a bit and I would love to get feedback to this one rather than the larger font.

I will push a new beta from release in an hour.

**Comment by eddiemuc** [comment]
If you push a beta right now I would like to include #11148 (adds simplified status filter to basic filter). I was hoping to discuss this tomorrow first, but I think this is really useful regardless of the outcome of filter discussion.

**Comment by eddiemuc** [comment]
...I just resolved merge conflict for #11148 (waiting for CI to build)

**Comment by Lineflyer** [comment]
Sorry, to late...but its OK. We can then still discuss tomorrow and push a new one in the next week with a complete solution.

**Comment by eddiemuc** [comment]
ah well, that's ok too :-)

**Comment by MagpieFourtyTwo** [comment]
Can we perhaps please solve #11081 "Remove "Remove list" question dialog" prior to next beta (cause it's really annoying) and perhaps even #11174 "Lists: Increase font brightness and contrast for Favo count in Rating field " (cause it does not really look good)?

**Comment by moving-bits** [comment]
I give you 50% :-)
#11174 is already fixed with #11152 and will be included, whereas for #11081 we have no easy solution found yet.

**Comment by Lineflyer** [comment]
As mentioned in #11242 a new beta version 2021.07.17-RC2 is uploaded to Google Play.
It includes everything up to 71a24f6e7cc42e87286d2c7a183787113cd1e5e9

**Comment by Lineflyer** [comment]
Just a quick status update:

Support mail is relatively quiet after the first text style improvements are contained in current beta.
Only quite some people who are missing their transparent background....but (as discussed before) it makes no sense. Maybe if we have kind of a dashboard in bottom navigation we could think about a custom background for that.

IMHO we could go into another (last?) beta round once #11273 is ready and tested.

**Comment by MagpieFourtyTwo** [comment]
Although >100 people are currently running the beta ... sounds pretty great.
Nevertheless, lets wait for the next weekend, when these guys are out in the field ... ;)
BTW: New filter system is already included in RC, too?

**Comment by eddiemuc** [comment]
> BTW: New filter system is already included in RC, too?

Oh yes :-)

**Comment by MagpieFourtyTwo** [comment]
So they are perhaps just shocked, thus speechless ...? ;)

**Comment by Lineflyer** [comment]
The only "regression" still reported are the functionless global hiding settings.
But here we are on a very good path, already having the filter-split and the connection between quick settings and filter. This is not yet contained in the newest RC as I wanted to wait for the migration first (#11273).

**Comment by eddiemuc** [comment]
@Lineflyer migration for the global "exclude" flags is included (although there might be a bug, add report of @MagpieFourtyTwo ), only migration of global cache type is missing.

**Comment by Lineflyer** [comment]
@eddiemuc But then we completely misunderstood each other in #11273, did we?

**Comment by eddiemuc** [comment]
> @eddiemuc But then we completely misunderstood each other in #11273, did we?
> 

Hm after rereading #11273, I indeed misunderstood your request, although you formulated it pretty clear. Sorry for that.

Anyway, right now the migration of the exclude own/found/dis/rc settings is done (with potential bug as said), but migration of global cache type filter is missing.

**Comment by fm-sys** [comment]
>but migration of global cache type filter is missing. 

Not sure, whether we should migrate the global cache type filter at all. Maybe it would be wise to let users restart from scratch. I know quite some "weekend cacher" who had set the global cache type filter at some point in the past but cannot remember anymore at all. While mysteries are kind of "hard stuff", e.g. multis can be interesting for them as well. As they now can select more than only one cache type it might be interesting for them if they don't get restricted automatically but can review their choice. 

But I don't have a real opinion on this...

**Comment by fm-sys** [comment]
Oh, just saw that this is the same as already agreed in #11273...

**Comment by Lineflyer** [comment]
Now that the cache hiding migration is working, we could anytime go for a next/last beta round.
There are some open PRs though...merge as needed, I will probably prepare a new beta tonight.

**Comment by eddiemuc** [comment]
Merged #11315 so it becomes part of tonights beta

**Comment by murggel** [comment]
If that will be the last beta round, #11296 (compact D/T filter) needs to get merged (I try to finish it today), because removing the single difficulty and terrain filter will needs migration otherwise.

**Comment by eddiemuc** [comment]
> If that will be the last beta round, #11296 / #11180(compact D/T filter) needs to get merged (I try to finish it today), because removing the single difficulty and terrain filter will needs migration otherwise. Or does still for release the old filter get lost?

I agree with @murggel . This should definitely go to production to spare us complex and unnecessary migraine efforts later.

**Comment by murggel** [comment]
Unfortunately I am back on my machine in approx 3 hours, then I can push the remaining changes to remove the single filters. Don't know if then is sufficient time for merging...

**Comment by eddiemuc** [comment]
> Unfortunately I am back on my machine in approx 3 hours, then I can push the remaining changes to remove the single filters. Don't know if then is sufficient time for merging...

I rechecked the code. With the exception of the missing "removal code", everything looks good code-wise (even the connectors oc and gc are changed!). I see several options on how to proceed:
* Merge without the removal changes. @murggel as far as I can tell from code, this would work. I think you even made sure that the old filters TERRAIN and DIFFICULTY are no longer selectable in the advanced filter type list - correct? So we could make that cleanup later
* Wait for murggel to return home and upload the changes. @Lineflyer what time are you planning to build the new beta
* Postphone the beta by a few days, giving some time to solve this in a more calm way and give the chance to retest it in the nightly before it goes to beta

I would prefer the third option. What do you say?

**Comment by murggel** [comment]
I would also prefer the third option 😊

**Comment by murggel** [comment]
With the first option probably the issue with the long click fits also in the release, there I definitely need one more day

**Comment by murggel** [comment]
> @murggel as far as I can tell from code, this would work. I think you even made sure that the old filters TERRAIN and DIFFICULTY are no longer selectable in the advanced filter type list - correct? So we could make that cleanup later

Correct,  thought, it is only on my local machine, but I did already that push 😊

**Comment by eddiemuc** [comment]
Ah in this case I will merge. Thanks @murggel for this contribution

**Comment by murggel** [comment]
Can someone test it before it's going into beta?

**Comment by eddiemuc** [comment]
I tested on my local emulator and it works fine

**Comment by murggel** [comment]
> I tested on my local emulator and it works fine

Thanks

**Comment by Lineflyer** [comment]
No need to rush. If you tell me you need time, we can wait a day or two.

**Comment by eddiemuc** [comment]
Now we're ready for a beta from my perspective. With a little more waiting time we may also include " long click filter select" (#11202), but we may also do another beta for that

**Comment by MagpieFourtyTwo** [comment]
I would also vote to wait, to not overstress the beta testers with too many versions (plus the immediate change of the D/T filter). Moreover, although the long click filter selection is a really helpful feature, it would most probably not really warrant its own beta ...?

However - the more time to test, the less fixes needed, the sooner it will become a real new playstore version.

Just my 2 cents ;)

**Comment by murggel** [comment]
I hope I need not too much time for the long click...

**Comment by Lineflyer** [comment]
As said...take your time.
Now that you said, there is more to come we should wait...releasing today was only an idea now that migration of hiding should work.

However now quite some changes are in, so we should provide a few days for testing/using in nightlies.
Lets see middle of this week for example.

**Comment by Lineflyer** [comment]
At this moment #11332 is blocking for a next beta release

**Comment by moving-bits** [comment]
> At this moment #11332 is blocking a next beta release

fixed with #11333

**Comment by fm-sys** [comment]
Don't know whether it would make sense to wait for  #11288 before releasing the first beta with two filter contexts. That would probably save us quite some migration routines. OTOH, as no one is currently working on that issue we can probably also release the beta before...

**Comment by Lineflyer** [comment]
I think we wait anyway for the CI preconditions regarding the fix for #11001. So we aill take our time.
I need to check #11288 though.

**Comment by Lineflyer** [comment]
I guess solving #11288 needs more time though. It can be part of a next release maybe.
I have however one question for that, but will place it into that issue.

Besides that, we do now have the preconditions for #11001 met so it should work again for an upcoming beta version.
This fact and the fact, that migration of the global hiding options + many fixes are already in, makes me think its time for the next/maybe final beta tonight?

**Comment by moving-bits** [comment]
#11349 should be included, IMHO. If the colors are ok for you, please merge (or drop me a hint), otherwise I can provide a differentiation between dark and light mode.

**Comment by Lineflyer** [comment]
#11349 merged for now (and cherry-picked to release as it targeted `master`). 
Will prepare another RC round now....shout out stop here to stop me ;)

**Comment by Lineflyer** [comment]
2021.07.28-RC is out on Google Play. May need some hours until available.

**Comment by Lineflyer** [comment]
Due to the fix #11361 I would like to push another beta update today.
Any objections to include #11359?

**Comment by Lineflyer** [comment]
I reverted the merge of #11359 as the feature is not working (due to reasons outside the code).
I will push another beta now with this state.

Regarding #11359 we should take some days to think and decide. It was maybe a bit short termed to merge this.

**Comment by Lineflyer** [comment]
2021.07.30-RC briefly smoke tested and pushed to beta testers.

**Comment by fm-sys** [comment]
In https://github.com/cgeo/cgeo/pull/11372#issuecomment-890450249 the question came up how much work needs to be done before we are ready for the next playstore release.

Can we collect a "blockers list" or create a milestone for it, so that we can organize the "last meters"? 

We definitely need to stop introducing new features to `release` but rather target them to `master`. As the bottom navigation is not yet ready anyway, we can think about doing another small beta/release cycle directly after the "big change" is out.

**Comment by Lineflyer** [comment]
> We definitely need to stop introducing new features to `release` but rather target them to `master`. 

I 100% agree. But didn't we do that already? I have to admit I did not follow all commits, but once we are in a beta phase we should not introduce new features on `release` for sure but focus on bugfixes.

Regarding the blocking points:
At this moment the only thing coming to my mind is #11204. But for this (as also for the other crash reports I filed into issues) its difficult to say how frequent those crashed appear to users.

**Comment by eddiemuc** [comment]
I am looking into those issues currently, specifically #11204. I agree we should focus on those only in release until we finally get the new version into production

**Comment by Lineflyer** [comment]
I packed all issues which I think are related into a milestone collection.
Not each of those need to be solved, but maybe looked at it to judge whether those are critical and/or easily solvable:
https://github.com/cgeo/cgeo/milestone/94

**Comment by Lineflyer** [comment]
BTW: What about this "underlying" error...no crash....but also the corresponding menu is still no longer attached to the actionbar.

**Comment by Lineflyer** [comment]
I removed #11345 from the "ToDo" milestone as this is a feature improvement rather than a bugfix.
What about the other items in https://github.com/cgeo/cgeo/milestone/94 still shown as open. For some of them there have been related commits merged. Can some be closed already?

In any case I think we should do a final beta round once we have aligned, that all known "ToDo" are done.

**Comment by eddiemuc** [comment]
While #11345 is not strictly a bug, it would be dramatically easy to solve imho. I will take a look.

#11204 might or might not be fixed by PR #11385. I fear we have to roll it out into beta and see whether the error persists.

For #11184, a fix was provided by @moving-bits 

This leaves #11351...

**Comment by Lineflyer** [comment]
If we find a fix for #11411 (and maybe for the other leftovers in https://github.com/cgeo/cgeo/milestone/94 ) I would vote for another (short) round of beta to see how the fixes behave in the field.

**Comment by Lineflyer** [comment]
I do see only two items left on the milestone list.
[x] #11184 - It seems it could be already solved by commit a18cdae90667ebd749890ff3918830e9520acc85 (to be verified during beta)
[ ] #11411 -> Still open

So I guess one more issue to fix to run a final beta round?

**Comment by fm-sys** [comment]
Another very valid change request came in just this morning, which probably should be tackled before our final release as well. See  #11430

**Comment by fm-sys** [comment]
[Here](https://github.com/cgeo/cgeo/pull/11431) is a fix for #11411. 

But we should wait for #11430. @eddiemuc to you have time + interest to work on it?

---

Furthermore, we have quite some PR's open targeting to release, which should be merged before the last beta. See:

https://github.com/cgeo/cgeo/pulls?q=is%3Apr+base%3Arelease+is%3Aopen+draft%3Afalse

**Comment by moving-bits** [comment]
What about changelog strategy (#11101)? - publishing a new release would be a good starting point...

**Comment by murggel** [comment]
> But we should wait for #11430. @eddiemuc do you have time + interest to work on it?

If @eddiemuc does not have time, I can have a look at #11430

**Comment by murggel** [comment]
PR #11434 fpr #11430 is on its way

**Comment by eddiemuc** [comment]
> > But we should wait for #11430. @eddiemuc do you have time + interest to work on it?
> 
> If @eddiemuc does not have time, I can have a look at #11430

@murggel thanks a lot for taking this!

**Comment by Lineflyer** [comment]
As just discussed in our meeting I will push a new beta with current state of `release` tonight (later) or tomorrow.

**Comment by moving-bits** [comment]
Since lab adventures are back to the game: Shouldn't we agree on a conclusion for #10911 before releasing (not necessarily before beta)?

**Comment by Lineflyer** [comment]
Looking at Google Play Console and support mail it looks like the latest beta is running quite smooth.

Regarding #10911:
I think its a very minor issue...still: If we do not solve it at any point of time it might get forgotten.
But as there is a need for a DB migration/version upgrade behind I am a bit reluctant to bring this in immediately before release. Shall we put it on `release` after releasing have it in the next bugfix release?
Or is there a significant disadvantage associated with that?

**Comment by Lineflyer** [comment]
Addition:
As beta looks good I suggest to finally release it early next week.

**Comment by fm-sys** [comment]
> As beta looks good I suggest to finally release it early next week.

In https://github.com/cgeo/cgeo/discussions/10359#discussioncomment-1155672:
> I scheduled a posting for 16.08.2021 11:00 UTC+02 using the logo and the text above on Facebook.

Maybe we can also publish the release on that day as well?

Then we could add some hint about the new release to the Facebook posting... "Birthday present release" ;-) 

---

> Regarding #10911:
I think its a very minor issue...still: If we do not solve it at any point of time it might get forgotten.
But as there is a need for a DB migration/version upgrade behind I am a bit reluctant to bring this in immediately before release. Shall we put it on release after releasing have it in the next bugfix release?

IMHO it's not a bug, but a change. I would suggest to fix #11430 and  #10911 on `master`, and start another very small feature release directly after the UI release runs smoothly. We have some (although really not many) cool features laying on `master` for months now, waiting to be finally released. As long as no urgent bugs occur, we probably don't need a separate bugfix release in between... (scrum approach)

**Comment by moving-bits** [comment]
> I would suggest to fix #11430 and #10911 on `master`, and start another very small feature release directly after the UI release runs smoothly.

ok for me as well

**Comment by Lineflyer** [comment]
So...lets roll the release?!
If no one stops I will brew the release and publish tonight. So it will be ready as 10 year c:geo birthday present to users.

@fm-sys Thanks for the nice birthday header!

Release checklist:

- [ ] Check/Insert correct version name as headline in ```/main/res/raw/version_release.txt``` 
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid 
- [ ] Trigger notification to status.cgeo.org via CI server once the release is available on Google Play
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by Lineflyer** [comment]
OK, release is pushed to Google Play. Perfect match for our 10th anniversary.

**Big thank you** to all involved for that great piece of work!! Fantastic!

**Comment by Lineflyer** [comment]
Note:
Strangely Google Play shows, that release 2021.08.15 is available but at the same time shows 2021.08.08-RC is still on beta channel.
Normally there used to be a remark like "Replaced by production". Not sure, why this does not show up this time.

If any of the devs happens to be beta tester, you might check whether you get 2021.08.15 delivered from Google Play.

**Comment by DaTechniker** [comment]
Happy Birthday.
For me it looks fine. 
Beta/RC was updated to 2021.08.15
Nice Design in Release Notes also :-) 
Thanks for the good work

![Screenshot_20210815-231153_cgeo](https://user-images.githubusercontent.com/88573756/129493123-d2e1ba34-9e51-46f6-af8a-077044dcadfb.jpg)
![Screenshot_20210815-230648_Google Play Store](https://user-images.githubusercontent.com/88573756/129493124-f60654a7-d711-4a9b-aeba-40f8bf6ceaff.jpg)

**Comment by eddiemuc** [comment]
> If any of the devs happens to be beta tester, you might check whether you get 2021.08.15 delivered from Google Play.

My Beta got updated by Google Play to new release as well. Looks really nice. Thanks for all the hard work!

**Comment by Narrat** [comment]
Hands on the new version. And yes, looks great :) and I'm already waiting on the further redesign :D
But one thing that doesn't look so nice, imo: even in a small list (55 Caches), without filtering (read that issue) it takes some time until the details are opened and it is visibly switching from waypoints to overview, which is somewhat odd to look at.

**Comment by Lineflyer** [comment]
Thanks for your feedback, seems its working then.

@Narrat  I took your problem into a dedicated issue --> #11476 

Done here..Hurray!
